src/envoy/http/authn/jwt_authn_store.h

```diff
@@ -0,0 +1,82 @@
+/* Copyright 2018 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "common/common/logger.h"
+#include "envoy/server/filter_config.h"
+#include "envoy/thread_local/thread_local.h"
+#include "src/envoy/http/jwt_auth/auth_store.h"
+#include "src/envoy/http/jwt_auth/config.pb.h"
+
+namespace Envoy {
+namespace Http {
+namespace IstioAuthn {
+
+enum JwtStoreType { PEER_STORE = 0, ORIGIN_STORE = 1 };
+
+typedef std::map<JwtStoreType,
+                 std::vector<Envoy::Http::JwtAuth::JwtAuthStoreFactory>>
```

Comment:
I think this should be JwtAuthStore, not factory.

Reply:
Because JwtAuthStoreFactory has included the thread local related implementations whereas JwtAuthStore does not have thread local related implementations, this should be a map of JwtAuthStoreFactory, instead of a map of JwtAuthStore.